src/components/Details/Details.js

```diff
@@ -0,0 +1,135 @@
+import React, { useCallback, useEffect, useState } from 'react';
+import {
+  ScrollView,
+  Text,
+  ImageBackground,
+  View,
+  Image,
+  ActivityIndicator,
+} from 'react-native';
+import { useTheme } from '@react-navigation/native';
+import { PropTypes } from 'prop-types';
+
+import styles from './styles';
+import Center from 'components/Center';
+import { fetchGenres } from 'controllers/MoviesClient';
+import MovieInfo from 'components/MovieInfo/MovieInfo';
+import textStyles from 'helpers/TextStyles';
+
+const Details = ({ route }) => {
+  const [genres, setGenres] = useState([]);
+  const [isLoading, setIsLoading] = useState(false);
+  const { colors } = useTheme();
+  const {
+    prefixUrl,
+    movie: {
+      backdrop_path: backdropPath,
+      poster_path: posterPath,
+      title,
+      original_title: originalTitle,
+      overview,
+      release_date: releaseDate,
+      vote_average: voteAverage,
+      genre_ids: genreIds,
+    },
+  } = route.params;
+
+  const getGenresNames = useCallback((ids, resGenres) => {
+    return ids && resGenres
+      ? ids.map(id => {
+          const foundGenre = resGenres.find(genre => genre.id === id);
+          return foundGenre ? foundGenre.name : null;
+        })
+      : [];
+  }, []);
+
+  const fetchGenresCallback = useCallback(async () => {
+    setIsLoading(true);
+    const res = await fetchGenres();
+    const genresNames = getGenresNames(genreIds, res.genres);
+    setGenres(genresNames);
+    setIsLoading(false);
+  }, [setGenres, getGenresNames, genreIds]);
+
+  useEffect(() => {
+    fetchGenresCallback();
+  }, [fetchGenresCallback]);
+  const uri = prefixUrl + backdropPath;
+  const thumbUri = prefixUrl + posterPath;
+
+  return (
+    <>
+      {isLoading && (
+        <Center>
+          <ActivityIndicator />
+        </Center>
+      )}
```

Comment:
What do you think about putting this indicator only in the "genres" section? Since all the other information is already there, you could show it and then only update that section